### PR TITLE
adapt tidb/pd/tikv arguments for v4.0.3

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 
adapt tidb/pd/tikv arguments for v4.0.3

### What is changed and how does it work?
1. add backup config for tikv, https://github.com/tikv/tikv/pull/8199
2. add `enable-telemetry` and deprecate `disable-telemetry` for pd, related PR: https://github.com/pingcap/pd/pull/2654

### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test

Related changes

 - Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:

```release-note
add `backup` config for tikv server, add `enable-telemetry` and deprecate `disable-telemetry` config for pd server
```

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/pd_config.go [hunk 1]
```diff
@@ -139,7 +144,26 @@ type DashboardConfig struct {
 	// When not disabled, usage data will be sent to PingCAP for improving user experience.
 	// Optional: Defaults to false
 	// +optional
+	//+k8s:openapi-gen=false
```
Comment (cofyc) [inline pkg/apis/pingcap/v1alpha1/pd_config.go:147]: add a deprecated notice in the comment, g.g. `Deprecated in PD v4.x.x, use EnableTelemetry instead`

please update the `v4.x.x` to the correct version in which this field is deprecated.

Comment (cofyc) [reply pkg/apis/pingcap/v1alpha1/pd_config.go:147]: If the PD plans to remove this field in the future, please add the removal version too. For example, `Deprecated in PD v4.x.x, planned for removal in vx.y.z`

Comment (lichunzhu) [reply pkg/apis/pingcap/v1alpha1/pd_config.go:147]: deleted in https://github.com/pingcap/tidb-operator/pull/2964/commits/46c14c678a2bf9492f21a6b34e8f26b0d0f75822

File: pkg/manager/member/tikv_member_manager_test.go [hunk 2]
```diff
@@ -2196,6 +2196,43 @@ func TestTransformTiKVConfigMap(t *testing.T) {
...
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			tc := newTidbClusterForTiKV()
+			tc.Spec.TiKV.Config.Backup = &v1alpha1.TiKVBackupConfig{
+				NumThreads: pointer.Int64Ptr(test.numThreads),
+			}
+			confText, err := MarshalTOML(tc.Spec.TiKV.Config)
+			g.Expect(err).NotTo(HaveOccurred())
+			g.Expect(test.result).Should(Equal(transformTiKVConfigMap(string(confText), tc)))
```
Comment (cofyc) [inline pkg/manager/member/tikv_member_manager_test.go:2231]: `transformTiKVConfigMap` is used to fix an incompatible field in different TiKV versions. No need to call this function here. I would suggest removing this function and update this test to verify the marshaled toml string only.

File: pkg/manager/member/tikv_member_manager_test.go [hunk 3]
```diff
@@ -2196,6 +2196,43 @@ func TestTransformTiKVConfigMap(t *testing.T) {
 	}
 }
 
+func TestTransformTiKVBackupConfig(t *testing.T) {
```
Comment (cofyc) [inline pkg/manager/member/tikv_member_manager_test.go:2199]: ### 
```suggestion
func TestTiKVBackupConfig(t *testing.T) {
```

File: pkg/manager/member/tikv_member_manager_test.go [hunk 4]
```diff
@@ -2196,6 +2196,43 @@ func TestTransformTiKVConfigMap(t *testing.T) {
...
+			result: `[backup]
+  num-threads = 24
+`,
+		},
+		{
+			name:       "4.0.3",
+			numThreads: 24,
+			result: `[backup]
+  num-threads = 24
+`,
```
Comment (cofyc) [inline pkg/manager/member/tikv_member_manager_test.go:2219]: these two cases are the same, just keep one

**Comment by DanielZhangQD** [comment]
@lichunzhu Please add release notes and update doc for this PR.

**Comment by lichunzhu** [comment]
> @lichunzhu Please add release notes and update doc for this PR.

OK

**Pull request merged**